_posts/2024-XX-XX-quarkus-kiota.adoc

```diff
@@ -0,0 +1,160 @@
+
+[source,xml]
+----
+<plugin>
+  <groupId>io.quarkus</groupId>
+  <artifactId>quarkus-maven-plugin</artifactId>
+  <extensions>true</extensions>
+  <executions>
+    <execution>
+      <goals>
+        <goal>build</goal>
+        <goal>generate-code</goal>
+      </goals>
+    </execution>
+  </executions>
+</plugin>
+----
+
+Add the necessary dependencies, we are going to replace the default http and json serialization libraries published by Microsoft and swap in Vert.X and Jackson(published from this https://github.com/kiota-community/kiota-java-extra[repository]) as they nicely play with the rest of the stack of a typical Quarkus based application.
+
+[source,xml]
+----
+<dependency>
+  <groupId>com.microsoft.kiota</groupId>
+  <artifactId>microsoft-kiota-abstractions</artifactId>
+  <version>${kiota.libs.version}</version>
+</dependency>
+<dependency>
+  <groupId>io.kiota</groupId>
+  <artifactId>kiota-http-vertx</artifactId> <!-- alternatively <artifactId>kiota-http-jdk</artifactId> -->
+  <version>{kiota-java-extra.version}</version>
+</dependency>
+<dependency>
+  <groupId>io.kiota</groupId>
+  <artifactId>kiota-serialization-jackson-quarkus</artifactId>
+  <version>{kiota-java-extra.version}</version>
+</dependency>
+<dependency>
+  <groupId>com.microsoft.kiota</groupId>
+  <artifactId>microsoft-kiota-serialization-text</artifactId>
+  <version>${kiota.libs.version}</version>
+</dependency>
+<dependency>
+  <groupId>com.microsoft.kiota</groupId>
+  <artifactId>microsoft-kiota-serialization-form</artifactId>
+  <version>${kiota.libs.version}</version>
+</dependency>
+<dependency>
+  <groupId>com.microsoft.kiota</groupId>
+  <artifactId>microsoft-kiota-serialization-multipart</artifactId>
+  <version>${kiota.libs.version}</version>
+</dependency>
+<dependency>
+  <groupId>jakarta.annotation</groupId>
+  <artifactId>jakarta.annotation-api</artifactId>
+</dependency>
+----
+
+Now we need to generate the actual client for our OpenAPI description, to do so, you should drop the OpenAPI file(in ``yaml`` or ``json`` format) in the ``src/main/openapi`` folder of your project.
+You are all set to use the client in your application!
+
+[source,java]
+----
+var client = new ApiClient(new VertXRequestAdapter(vertx));
+client.
+----
+
+striking ``.`` after client, the code completion of your IDE kicks in and provide you a beautiful, fully typed, builder pattern matching the endopoint descriptions provided in the OpenAPI specification.
+
+image::completion.jpeg[Code Completion]
+
+For example an endpoint definition like https://github.com/Apicurio/apicurio-registry/blob/6882af10e9de8e1d245006db01f039b1fbf6355a/common/src/main/resources/META-INF/openapi-v2.json#L668[this one] nicely unroll in Java as:
+
+[source,java]
+----
+client
+  .groups()
+  .byGroupId(groupId)
+  .artifacts()
+  .byArtifactId(artifactId)
+  .meta()
+  .get();
+----
+
+== Alternatives
```

Comment:
I think it is kinda missing an answer to the question - ok, so this kiota thing now exist - what does it mean for me  as a quarkus user - what should I use it for ? when should i use quarkus rest client vs this and does the kiota approach give us the same level of integration with quarkus tracing,monitoring, failover, security etc.

might not know all the answers to that but lets at least guide users a bit here.